R/FilterStatesSE.R

```diff
@@ -337,6 +252,103 @@ SEFilterStates <- R6::R6Class( # nolint
     },
 
     # shiny modules ----
+    #' @description
+    #' Shiny module UI
+    #'
+    #' Shiny UI element that stores `FilterState` UI elements.
+    #' Populated with elements created with `renderUI` in the module server.
+    #'
+    #' @param id (`character(1)`)\cr
+    #'   shiny element (module instance) id
+    #'
+    #' @return `shiny.tag`
+    #'
+    ui = function(id) {
+      ns <- NS(id)
+      datalabel <- self$get_datalabel()
+      tags$div(
+        class = "list-group hideable-list-group",
+        `data-label` = ifelse(datalabel == "", "", datalabel), # todo: labels are not displayed for MAE - see filter-panel.css
+        shiny::tagList(uiOutput(ns("genes"))),
+        shiny::tagList(uiOutput(ns("samples")))
+      )
+    },
+
+    #' @description
+    #' Shiny server module.
+    #'
+    #' @param id (`character(1)`)\cr
+    #'   shiny module instance id
+    #'
+    #' @return `moduleServer` function which returns `NULL`
+    #'
+    server = function(id) {
```

Comment:
Is the reason we need this only here  because self$state_list_get(1) works for other filterstate classes but there are two queues here so we need a different server? I'm a little lost tbh